node/src/components/linear_chain_sync/operations.rs

```diff
@@ -484,6 +472,26 @@ where
         }
     }
 
+    // Fetch and store all blocks that can contain not-yet-expired deploys. These are needed for
+    // replay detection.
+    let mut current_header = most_recent_block_header.clone();
+    while current_header.timestamp().elapsed() < chainspec.deploy_config.max_ttl
```

Comment:
(Not your fault — mine, actually!)
I wonder whether this should be

```suggestion
    while era_start.saturating_diff(current_header.timestamp()) < chainspec.deploy_config.max_ttl
```

instead, where `era_start` is the start of the most recent era we know of?

Because while we'll only _propose_ blocks with future timestamps, we might have to validate proposals with a timestamp older than that, but never older than the beginning of the most recent era.

Not sure if we have that era start information readily available here — otherwise it should probably be a separate ticket.

Reply:
Looks like this is a blocker; `era_start` here is the trusted-key-block timestamp.

Reply:
Fixed in fd7afac00bc4bef26ad730b1a74fddd3908e7c27